htcanalyze/htcanalyze.py

```diff
@@ -1173,7 +1176,7 @@ def gen_time_dict(submission_date: date_time = None,
     Depending on the given arguments,
     this function will try to calculate
     the time differences between the events.
-    If not all three values are given, it will return None
+    If not all three values are given, it will return None => then why don't you check first if any is None and immediately return None?
```

Comment:
This additional comment should be put as a TODO on its own line or (even better) implement the suggested improvement in the code.